src/ipnd/services.rs

```diff
@@ -0,0 +1,287 @@
+                    let first: f32 = input[0].parse().expect("Couldn't parse latitude");
+                    let second: f32 = input[1].parse().expect("Couldn't parse longitude");
+                    let mut bytes = first.to_be_bytes().to_vec();
+                    bytes.extend(second.to_be_bytes().to_vec().iter());
+                    Ok((tag, bytes))
+                }
+            },
+            // 191 expect an integer between 0 and 100 to represent battery level in %
+            191 => {
+                let res = payload.parse::<i8>();
+                if let Ok(input) = res {
+                    if input > 100 || input < 0 {
+                        Err(String::from("Provided number can not be used to represent battery level. Please provide a number between 0 and 100"))
+                    } else {
+                        Ok((tag, input.to_be_bytes().to_vec()))
+                    }
+                } else {
+                    Err(String::from(format!("Could not parse provided argument into an integer. {}", res.expect_err(""))))
+                }
+            },
+            // 255 expects 5 arguments String Int Int String String to represent an address
+            255 => {
+                let input: Vec<&str> = payload.split_whitespace().collect();
+                if input.len() == 5 {
+                    Ok((tag, payload.as_bytes().to_vec()))
+                } else {
+                    Err(String::from("Can not derive address from provided arguments. Argument order is: Street HouseNumber PostalNumber City CountryCode"))
+                }
+            },
+            // Undefined tags
+            64..=126 | 128..=190 | 192..=254 => {
+                Err(String::from("This custom tag is not yet defined. Please refrain from using it until added."))
+            }
+        }
+    }
+
+    /// Check if the ServiceBlock contains no CLAs and no other Services
+    pub fn is_empty(&self) -> bool {
+        self.clas.len() + self.services.len() == 0
+    }
+}
+
+// Implementation of the Display trait for ServiceBlocks for proper formatting
+impl std::fmt::Display for ServiceBlock {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut output = String::new();
+        output.push_str("ConvergencyLayerAgents:\n");
+        let mut counter = 0;
+        for (name, port) in self.get_clas() {
+            let str = if port.is_some() {
+                format!(
+                    "{}. CLA: Name = {} Port = {}\n",
+                    counter,
+                    name,
+                    port.unwrap()
+                )
+            } else {
+                format!("{}. CLA: Name = {}\n", counter, name)
+            };
+            output.push_str(str.as_str());
+            counter += 1;
+        }
+        counter = 0;
+        output.push_str("Other services:\n");
+        for (tag, payload) in self.get_services() {
+            let str = match tag {
+                63 => format!(
```

Comment:
same with the numbers as above.. if one changes then it has to be fixed in several places